qdrant-landing/content/articles/binary-quantization.md

```diff
@@ -0,0 +1,236 @@
+
+As is the case with other quantization methods, these benefits come at the cost of recall degradation. However, our implementation lets you balance the tradeoff between speed and recall accuracy at time of search, rather than time of index creation. 
+
+The rest of this article will cover:
+1. The importance of binary quantization
+2. Basic implementation using our Python client
+3. Benchmark analysis and usage recommendations
+
+![Speed vs recall of quantization and indexing methods](/articles_data/binary-quantization/bq-1.png)
+
+## What is Binary Quantization?
+Binary quantization (BQ) converts any vector embedding of floating point numbers into a vector of binary or boolean values. This feature is an extension of our past work on [scalar quantization](/articles/scalar-quantization/) where we convert `float32` to `uint8` and then leverage a specific SIMD CPU instruction to perform fast vector comparison. 
+
+![What is binary quantization](/articles_data/binary-quantization/bq-2.png)
+
+**This binarization function is how we convert a range to binary values. All numbers greater than zero are marked as 1. If it's zero or less, they become 0.**
+
+The benefit of reducing the vector embeddings to binary values is that boolean operations are very fast and need significantly less CPU instructions. In exchange for reducing our 32 bit embeddings to 1 bit embeddings we can see up to a 40x retrieval speed up gain!
+
+One of the reasons vector search still works with such a high compression rate is that these large vectors are over-parameterized for retrieval. This is because they are designed for ranking, clustering, and similar use cases, which typically need more information encoded in the vector. 
+
+For example, The 1536 dimension OpenAI embedding is worse than Open Source counterparts of 384 dimension at retrieval and ranking. Specifically, it scores 49.25 on the same [Embedding Retrieval Benchmark](https://huggingface.co/spaces/mteb/leaderboard) where the Open Source `bge-small` scores 51.82. This 2.57 points difference adds up quite soon.
+
+Our implementation of quantization achieves a good balance between full, large vectors at ranking time and binary vectors at search and retrieval time. It also has the ability for you to adjust this balance depending on your use case.
+
+## Fast Search and Retrieval
+
+Unlike product quantization, binary quantization does not rely on reducing the search space for each probe. Instead, we build a binary index that helps us achieve large increases in search speed.
+
+![Speed by quantization method](/articles_data/binary-quantization/bq-3.png)
+
+HNSW is the approximate nearest neighbor search. This means our accuracy improves up to a point of diminishing returns, as we check the index for more similar candidates. In the context of binary quantization, this is referred to as the **oversampling rate**. 
```

Comment:
I guess we can explain oversampling using simpler words. Possible idea:
1. First, add please the scoring definition. I asked about it above. We have to show that scores are very approximate
2. Second, explain about rescoring. We can use original vector data for top neighbors distance recalculation
3. Next, we are close to oversampling. Define the problem of accuracy and an idea of solution: search more points
4. Explain that just searching more is not a perfect solution: we have to rescore it
5. And only after that oversampling is clear: it takes more vectors for search for lower accuracy compensation